app.js

```diff
@@ -225,54 +205,62 @@ class Scrummy {
    *   The message from the client.
    * @return {undefined}
    */
-  revokeVote(data) {
-    if (!this.bucket[data.game]) {
-      throw new Error(`${data.game} does not exist!`);
+  revokeVote({ game: gameId, nickname }) {
+    const game = this.bucket.get(gameId);
+    if (!game) {
+      throw new Error(`${gameId} does not exist!`);
     }
     // If user has not voted
-    if (!this.bucket[data.game].votes[data.nickname]) {
-      throw new Error(`${data.nickname} has no votes to revoke!`);
+    if (!game.votes[nickname]) {
+      throw new Error(`${nickname} has no votes to revoke!`);
     }
-
-    delete this.bucket[data.game].votes[data.nickname];
-
-    Scrummy.broadcast(JSON.stringify({
-      type: 'clientRevoke',
-      data: { nickname: data.nickname },
-    }), this.bucket[data.game].clients);
-    logger(`${data.nickname} revoked his or her vote in ${data.game}\n`);
+    game.revokeVote({ nickname });
+    logger(`${nickname} revoked his or her vote in ${gameId}\n`);
   }
-
   /**
    *   Disconnects a client from the given game.
    *
    * @param {Object} data
    *   The message from the client.
    * @return {undefined}
    */
-  disconnect(data) {
-    if (!this.bucket[data.game]) {
-      throw new Error(`${data.game} does not exist!`);
+  disconnect({ game: gameId, nickname }) {
+    const game = this.bucket.get(gameId);
+    if (!game) {
+      throw new Error(`${gameId} does not exist!`);
     }
     // If user isn't a part of the game
-    if (!this.bucket[data.game].users.filter(user => user.nickname === data.nickname).length) {
-      throw new Error(`${data.nickname} is not a part of ${data.game}!`);
+    if (!game.hasUser({ nickname })) {
+      throw new Error(`${nickname} is not a part of ${gameId}!`);
     }
     // Filter out the user that disconnected from the user list.
-    this.bucket[data.game].users = this.bucket[data.game].users
-      .filter(user => user.nickname !== data.nickname);
+    game.users = game.users
+      .filter(user => user.nickname !== nickname);
     // Filter out the client reference for the user that disconnected from the user list.
-    this.bucket[data.game].clients = this.bucket[data.game].clients
-      .filter(user => user.nickname !== data.nickname);
+    game.clients = game.clients
+      .filter(user => user.nickname !== nickname);
     // Remove any votes cast by disconnected user.
-    delete this.bucket[data.game].votes[data.nickname];
+    game.revokeVote({ nickname });
 
-    Scrummy.broadcast(JSON.stringify({
+    game.broadcast(JSON.stringify({
       type: 'clientDisconnect',
-      data: { nickname: data.nickname },
-    }), this.bucket[data.game].clients);
-    logger(`${data.nickname} disconnected\n`);
+      data: { nickname },
+    }));
+    logger(`${nickname} disconnected\n`);
   }
 }
 
+Object.defineProperty(Scrummy, 'EXPOSED_METHODS', {
```

Comment:
i'd be down to add immutable as a dependency. At the very least a comment would help understand intent for future contributors.